src/main/java/io/xcherry/core/client/Client.java

```diff
@@ -275,17 +268,21 @@ private String startProcessInternal(
         final ProcessStartConfig processStartConfig
     ) {
         final Process process = registry.getProcess(processType);
-        final ProcessOptions processOptions = process.getOptions() == null
-            ? ProcessOptions.builder(process.getClass()).build()
-            : process.getOptions();
 
         final ProcessExecutionStartRequest request = new ProcessExecutionStartRequest()
             .namespace(clientOptions.getNamespace())
             .processId(processId)
             .processType(processType)
             .workerUrl(clientOptions.getWorkerUrl())
             .startStateInput(clientOptions.getObjectEncoder().encodeToEncodedObject(input))
-            .processStartConfig(toApiModel(process.getPersistenceSchema(), processStartConfig));
+            .processStartConfig(
+                processStartConfig == null
+                    ? null
+                    : processStartConfig.toApiModel(
+                        process.getPersistenceSchema(),
+                        clientOptions.getDatabaseStringEncoder()
```

Comment:
Not sure database string encoder is a good name. It's a converter to convert object into part of the query for database to execute.

Reply:
On SDK side, we only convert the  column value from object to string, ans vice versa.